realm/realm-library/src/main/java/io/realm/Realm.java

```diff
@@ -784,6 +784,64 @@ private Scanner getFullStringScanner(InputStream in) {
     }
 
     /**
+     * Bulk insert of a list of unmanaged RealmObjects, this is generally faster than {@link #copyToRealm(Iterable)} since it
+     * doesn't return the inserted elements, and performs minimum allocations and checks.
+     * <p>
+     * Please note, this doesn't check if the object is already managed or not.
+     * Copying an object will copy all field values. Any unset field in the object and child objects will be
+     * set to their default value if not provided.
+     *
+     * @param objects RealmObjects to insert.
+     */
+    public void insert(Collection<? extends RealmModel> objects) {
+        configuration.getSchemaMediator().insertToRealm(this, objects);
+    }
+
+    /**
+     * Insert of unmanaged RealmObject, this is generally faster than {@link #copyToRealm(RealmModel)} since it
+     * doesn't return the inserted elements, and performs minimum allocations and checks.
+     * <p>
+     * Please note, this doesn't check if the object is already managed or not.
+     * Copying an object will copy all field values. Any unset field in the object and child objects will be
+     * set to their default value if not provided.
+     *
+     * @param object RealmObjects to insert.
+     */
+    public void insert(RealmModel object) {
+        Map<RealmModel, Long> cache = new IdentityHashMap<RealmModel, Long>();
+        configuration.getSchemaMediator().insertToRealm(this, object, cache);
+    }
+
+    /**
+     * Bulk insert or update of a list of unmanaged RealmObjects, this is generally faster than {@link #copyToRealmOrUpdate(Iterable)} since it
+     * doesn't return the inserted elements, and performs minimum allocations and checks.
+     * <p>
+     * Please note, this doesn't check if the object is already managed or not.
+     * Copying an object will copy all field values. Any unset field in the object and child objects will be
+     * set to their default value if not provided.
+     *
+     * @param objects RealmObjects to insert.
+     */
+    public void insertOrUpdate(Collection<? extends RealmModel> objects) {
+        configuration.getSchemaMediator().insertOrUpdateToRealm(this, objects);
+    }
+
+    /**
+     * Insert or update an unmanaged RealmObject, this is generally faster than {@link #copyOrUpdate(RealmModel, boolean)}} since it
+     * doesn't return the inserted elements, and performs minimum allocations and checks.
+     * <p>
+     * Please note, this doesn't check if the object is already managed or not.
```

Comment:
Specifically, this doesn't check if the object is already managed, so inserting a managed object will duplicate it unlike `copyToRealm`.